File: personalizatio-sdk/src/main/java/com/personalizatio/stories/models/elements/LinkElement.java
package com.personalizatio.stories.models.elements;

import androidx.annotation.NonNull;

import org.json.JSONObject;

public interface LinkElement extends Element {

    default String getLink(@NonNull JSONObject json) {
        var link = json.optString("link_android", "");
        if (link.isEmpty()) {
            link = json.optString("link", "");
        }
        return link;
    }

    String getLink();
}

File: personalization-sdk/src/main/kotlin/com/personalization/stories/models/elements/LinkElement.kt
package com.personalization.stories.models.elements

import org.json.JSONObject

interface LinkElement : Element {
    fun getLink(json: JSONObject): String {
        var link = json.optString("link_android", "")
        if (link.isEmpty()) {
            link = json.optString("link", "")
        }
        return link
    }

    val link: String
}
